### PR TITLE
Using an attributes file to populate samhainrc

### PR DESCRIPTION
**Pull request by munjeli**
Unpacked the attributes hash in the default recipe and write the samhainrc. Passes Kitchen Converge

### CODE REVIEW

File: .kitchen.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+driver:
+  name: docker
+
+provisioner:
+  name: chef_solo
```
Comment (RoboticCheese) [inline .kitchen.yml]: Generally, `chef_zero` is the preferable provisioner over `chef_solo`. It's closer in behavior to how Chef would normally be run.

File: attributes/default.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
+default['samhain']['Attributes']['file'].tap do |f|
+  f['/etc/mtab'] = true,
+  f['/etc/ssh_random_seed'] = true,
```
Comment (RoboticCheese) [inline attributes/default.rb]: These commas aren't necessary, and might cause issues with the attributes; these are all individual assignment statements, not elements in a hash.

File: recipes/default.rb [hunk 3]
```diff
@@ -1,9 +1,37 @@
...
+# Cookbook Name:: samhain
 # Recipe:: default
 #
 # installs and configures samhain for host integrity monitoring
 
 package 'samhain'
 
+def build_config
+  samhainrc = ''
+  node['samhain'].each do |k, v|
```
Comment (RoboticCheese) [inline recipes/default.rb]: Are there any future attributes we might need that are unrelated to the config file? If there'll be, for example, an attribute for whether to enable and run Samhain as a service, putting these things under `node['samhain']['config']` might do a better job of keeping them separated.

File: recipes/default.rb [hunk 4]
```diff
@@ -1,9 +1,37 @@
 #
-# Cookbook Name:: chef-samhain
+# Cookbook Name:: samhain
 # Recipe:: default
 #
 # installs and configures samhain for host integrity monitoring
 
 package 'samhain'
 
+def build_config
```
Comment (RoboticCheese) [inline recipes/default.rb]: Any idea what class this effectively defines the method in? Helper methods would normally go in their own helpers module in `libraries/helpers.rb` or somesuch (e.g. in the [httpd cookbook](https://github.com/chef-cookbooks/httpd/blob/master/libraries/helpers.rb); I don't know if there's any danger to defining them inline in a recipe or not.

File: recipes/default.rb (hunk 3 above)
Comment (munjeli) [inline recipes/default.rb]: Yup I can do that. I was wondering about it, actually. Also I'm not sure what to do with attributes in the default file that are not exposed - but I reckon we can add them if we need them. There are some options commented out.

**Comment by RoboticCheese** [comment]
:boat:

**Pull request merged**